### PR TITLE
Use the same math for dragging blocks out of horizontal and vertical flyouts.

### PR DESCRIPTION
**Pull request by rachel-fenichel**
Blockly version of https://github.com/LLK/scratch-blocks/pull/1000

### CODE REVIEW

**Comment by AnmAtAnm** [comment]
LGTM

**Comment by marisaleung** [comment]
LGTM

File: core/flyout_base.js [hunk 1]
```diff
@@ -752,3 +752,62 @@ Blockly.Flyout.prototype.reflow = function() {
...
+  // The offset in pixels between the flyout workspace's origin and the upper
+  // left corner of the injection div.
+  var flyoutOffsetPixels = this.workspace_.getOriginOffsetInPixels();
+
+  // The position of the old block in flyout workspace coordinates.
+  var oldBlockPosWs = oldBlock.getRelativeToSurfaceXY();
+
+  // The position of the old block in pixels relative to the flyout
+  // workspace's origin.
+  var oldBlockPosPixels = oldBlockPosWs.scale(this.workspace_.scale);
```
Comment (picklesrus) [inline core/flyout_base.js]: Is scale really the only thing to do to convert between workspace and pixel coordinates?  What happens when the flyout is scrolled? is getRelativeToSurfaceXY handling that part? or is it the sum below with flyoutOffsetPixels?

Comment (rachel-fenichel) [inline core/flyout_base.js]: Scrolling is handled with `targetWorkspace.getOriginOffsetInPixels()`

Comment (picklesrus) [inline core/flyout_base.js]: Got it. Thanks.

**Pull request merged**